crypto/pkcs12/p12_sbag.c

```diff
@@ -103,6 +113,55 @@ PKCS12_SAFEBAG *PKCS12_SAFEBAG_create_crl(X509_CRL *crl)
                                     NID_x509Crl, NID_crlBag);
 }
 
+PKCS12_SAFEBAG *PKCS12_SAFEBAG_create_secret(int type, int vtype, const unsigned char *value, int len)
+{
+    PKCS12_BAGS *bag;
+    PKCS12_SAFEBAG *safebag;
+
+    if ((bag = PKCS12_BAGS_new()) == NULL) {
+        PKCS12err(0, ERR_R_MALLOC_FAILURE);
+        return NULL;
+    }
+    bag->type = OBJ_nid2obj(type);
+
+    switch(vtype) {
+    case V_ASN1_OCTET_STRING:
+        {
+            ASN1_OCTET_STRING *strtmp = ASN1_OCTET_STRING_new();
+
+            if (strtmp == NULL) {
+                PKCS12err(0, ERR_R_MALLOC_FAILURE);
+                goto err;
+            }
+            /* Pack data into an octet string */
+            ASN1_OCTET_STRING_set(strtmp, value, len);
+            bag->value.other = ASN1_TYPE_new();
+            if (strtmp == NULL) {
```

Comment:
should this be `bag->value.other` ?

Reply:
I think this is valid question. The code does not make much sense as is and there is also missing return value check on ASN1_OCTET_STRING_set().

Reply:
I believe it should be assigned to 'other', but it's not a clear cut case. In the struct pkcs12_bag_st definition there's a comment:

https://github.com/openssl/openssl/blob/64e54bf5c6657bf423d3ba463f31095d598d94e7/crypto/pkcs12/p12_local.h#L41

Secret bags can contain data of any ASN.1 type but for this API only an octet string is accepted. I can see this being the most common use case - pack application-specific data into an octet string and indicate the type of this data using an OID in the 'type' argument.

Reply:
The thing that @paulidale meant is probably that the `strtmp == NULL` check should be instead `bag->value.other == NULL`. The strtmp is checked already before but there is no check for the `ASN1_TYPE_new()` failure.

Reply:
Fixed now